File: code-alongs/04-ruby/99. warmups/ruby_squares.js
//same as ruby version but in javascsript!

function getSquares(numArray) {
    let squares = []
    numArray.forEach(function(num) {
        if (Math.sqrt(num) % 1 === 0) {
            squares.push(num)
        }
    });
    uniq = [...new Set(squares)];
    return uniq.sort((a, b) => a - b)
}

getSquares([4, 1, 16, 1, 10, 35, 22])

var uniqSquares = [...new Set(squares)];

File: code-alongs/04-ruby/99. warmups/ruby_squares.rb
require 'pry'

def get_squares (num_arr)
    perfect_sqr_arr = [] 

    num_arr.each do |num|
        num_sqrt = Math.sqrt(num)
        if num_sqrt == num_sqrt.round
            perfect_sqr_arr << num
        end
    end

    perfect_sqr_arr.uniq!.sort
end

p get_squares [4, 1, 16, 1, 10, 35, 22]

# binding.pry
